search-app/src/cmr/search/results_handlers/atom_results_handler.clj

```diff
@@ -427,6 +429,7 @@
                                                      (get has-granules-map id false))))
                (when granule-counts-map
                  (x/element :echo:granuleCount {} granule-count))
+               (x/element :echo:hasVariables {} has-variables)
```

Comment:
It looks like this is different than has-granules and granule-counts. For those we don't always return the field. I think this is controlled by a parameter `include_has_granules` and `include_granule_counts` (I might have the wrong names).

I'm not convinced that we need an include_has_variables flag, but it is not consistent.

Reply:
The short answer is we should not have added the `include_has_granules` parameter in our api doc. I guess the reason we added that parameter is we have thought that include has granules would be an expensive operation and we don't want to do it for all searches unless it is requested, but we have since introduced caching to has_granules and that makes the operation no longer expensive and unnecessary to have the `include_has_granules` parameter.

`has_variables` is totally different, it is directly indexed into the collection elasticsearch index and it is as simple as any other regular granule search field. There is definitely no reason to introduce an `include_has_variables` parameter.

Reply:
I agree with @ygliuvt's assessment, and would like to propose something a bit further: the planned deprecation of the `include_has_*` (i.e., identifying when we want to deprecate them, updating the docs in that version to let users know it's going away, and identifying the version where they will actually be removed).